javascript/javascript1/week2/candyFunction.js

```diff
@@ -0,0 +1,64 @@
+let boughtCandyPrices=[];
+let price;
+const amountToSpend=Math.random() * 100;
+
+function addCandy(candyType, weight)
+{
+    
+    if(candyType=="Sweet")
+    {
+        price=0.5*weight;
+        boughtCandyPrices.push(price);
+    }
+    else if(candyType=="Chocolate")
+    {
+        price=0.7*weight;
+        boughtCandyPrices.push(price);
+    }
+    else if(candyType=="Toffee")
+    {
+        price=1.1*weight;
+        boughtCandyPrices.push(price);
+    }
+    else
+    {
+        candyType="Chewing Gum";
+        price=0.03*weight;
+        boughtCandyPrices.push(price);
+
+    }
+    return boughtCandyPrices.toString();
+}
+function canBuyMoreCandy()
+{
+    addCandy('sweet', 20); 
+    addCandy('chocolate' , 12) ;
+    addCandy ('toffee' , 10) ;
+    addCandy('chewing gum', 6);
+
+    let totalPrice=0;
+    n=0;
+    while(n<boughtCandyPrices.length){
```

Comment:
Good use of while, however I think a traditional `for` loop would be more ergonomic here.